draft-ietf-capport-architecture.xml

```diff
@@ -112,24 +112,40 @@
        principles:
       </t>
       <ul spacing="normal">
-        <li>Solutions SHOULD NOT require the forging of responses from DNS or
-            HTTP servers, or any other protocol.  In particular, solutions
-            SHOULD NOT require man-in-the-middle proxy of TLS traffic.</li>
-        <li>Solutions MUST operate at the layer of Internet Protocol (IP) or
-            above, not being specific to any particular access technology such
-            as Cable, WiFi or mobile telecom.</li>
-        <li>Solutions MAY allow a device to be alerted that it is in a
-            captive network when attempting to use any application on the
-            network.</li>
+        <li>Current captive portal solutions typically implement some variations
+            of forging DNS or HTTP responses.
+            Some reportedly perform man-in-the-middle (MITM) proxy of HTTPS in
+            order to forge reponses.
+            Captive Portal Solutions should not have to break any protocols or
+            otherwise act in the manner of an attacker.
+            Therefore, solutions MUST NOT require the forging of responses from
+            DNS or HTTP servers, or any other protocol; solutions MUST NOT
+            require man-in-the-middle proxy of TLS traffic.</li>
+        <li>To maximize universality and adoption, solutions MUST operate at the
+            layer of Internet Protocol (IP) or above, not being specific to any
+            particular access technology such as Cable, WiFi or mobile
+            telecom.</li>
+        <li>Current captive portal solutions that work by affecting DNS or HTTP
+            can only alert applications using those protocols that the network
+            is a captive network. It would be beneficial to alert any
+            application. Hence, solutions MAY allow a device to be alerted that
+            it is in a captive network when attempting to use any application on
+            the network.</li>
         <li>Solutions SHOULD allow a device to learn that it is in a captive
```

Comment:
This is really the same requirement as the previous one.  Could this be reduced to "Devices SHOULD be able to query the network to determine whether the device is captive."  And then following with information about not being coupled to using intercepted protocols or being forced to send sacrificial requests.

Reply:
Two points combined. Please review.